src/onegov/activity/upgrade.py

```diff
@@ -888,3 +888,13 @@ def add_invoice_item_organizer(context):
             'invoice_items',
             column=Column('organizer', Text, nullable=True)
         )
+
+
+@upgrade_task('Add attendee id to invoice item and rm')
+def add_attendee_id_to_invoice_item_and_rm(context):
+    if not context.has_column('invoice_items', 'attendee_id'):
+        context.operations.add_column('invoice_items', Column(
+            'attendee_id', UUID, ForeignKey("attendees.id"), nullable=True
+        ))
+
+    context.operations.drop_column('invoice_items', 'period_id')
```

Comment:
You should probably go one step further and initialize `attendee_id` based on the current value of `group`, i.e. do the old annoying query where you have to join `Attendee` and `InvoiceItem` based on the `Attendee.username`, `Invoice.user_id`, `User.id` and the `Attendee.name`.

If you get more than one attendee for the same InvoiceItem somewhere, you can just default to whichever one is first.

Reply:
The upgrade step should still populate `InvoiceItem.attendee_id` based on the old way of joining the tables. Otherwise none of the existing InvoiceItems will be linked properly and a new InvoiceItem will be created every time a new course is booked.